src/components/SignUp.js

```diff
@@ -1,74 +1,164 @@
+        email: Yup.string().email().required(),
+        password: Yup.string().min(8).max(20).required(),
+        repeatPassword: Yup.string()
+            .oneOf([Yup.ref('password'), null])
+            .required(),
+    });
+
+    
+
+    const handleClick = async (e) => {
+        e.preventDefault();
+        setError(false); 
+        setErrorMessage(''); 
+        const isValid = await signupSchema.isValid({
+            firstName: firstName,
+            lastName: lastName,
+            username: username,
+            email: email,
+            password: password,
+            repeatPassword: repeatPassword,
+        });
+
+        if (isValid) {
+            axiosInstance
+                .post('/auth/signup', {
+                    firstName: firstName,
+                    lastName: lastName,
+                    username: username,
+                    email: email,
+                    password: password,
+                    gender: gender,
+                })
+                .then((response) => {
+                    alert("Account has been created!");
+                    navigate('/auth/signin');
+                    console.log(response);
+                })
+                .catch((err) => {
+                    console.log(err);
+                    console.log(err.response);
+                    setErrorMessage(err.response.data.msg); 
+                    setError(true);
+                });
+        } else {
+            setError(true); 
+            setErrorMessage('One or more input is invalid');
+        }
+    };
+
     return (
         <Container id="main-container" className="d-grid h-100">
-            <Row>
-                <Col sm={7}>
-                    <img class="info-Banner" src={picLogScreen} alt="picLogScreen"/>
-                </Col>
-
-                <Col sm={4}>
+            
+            <div className = "split left">
+                <div className="centered">
+                    <img className="info-Banner" src={picLogScreen} alt="picLogScreen"/>
+                </div>
+            </div>
+              
+            <div className = "split right">
+                <div className="centered">
                     <Form>
                         <h1 className="header-css">Register Account</h1>
                         <h6>For the purpose of industry regulation, your details are required.</h6>
-  
-                    <Form.Group className="mb-3" controlId="form-Basic-Fullname">
-                        <Form.Label>Full Name</Form.Label>
-                        <Form.Control className="name-input-css" type="name" placeholder="Enter full name" />
+
+                    <Form.Group className="mb-3" controlId="formBasicFirstname">
+                        <Form.Label>First Name</Form.Label>
+                        <Form.Control className="name-input-css" type="name" placeholder="Enter first name" onChange={(e) => {setFirstName(e.target.value)}}/>
```

Comment:
Using one `useState` could have potentially reduced all your `onChange` handlers to one `onChange` handler